docs/development/authentication.md

```diff
@@ -0,0 +1,137 @@
+# Understanding authentication at runtime
+
+The following document provides an introduction around the different authentication methods that can take place during an image build when using the Build operator.
+
+- [Overview](#overview)
+- [Authentication for Git](#authentication-for-git)
+  - [Basic authentication](#basic-authentication)
+  - [SSH authentication](#ssh-authentication)
+  - [Usage of git secret](#usage-of-git-secret)
+- [Authentication to container registries](#authentication-to-container-registries)
+  - [Docker Hub](#docker-hub)
+  - [Usage of registry secret](#usage-of-registry-secret)
+- [References](#references)
+
+## Overview
+
+There are two places where users might need to define authentication when building images. Authentication to a container registry is the most common one, but also users might have the need
+to define authentications for pulling source-code from Git.
```

Comment:
I would remove the line-break and and extend your sentence:

*... source code from Git if a private Git repository is used*

Reply:
done